vunit/sim_if/ghdl.py

```diff
@@ -375,8 +381,14 @@ def simulate(  # pylint: disable=too-many-locals
             gcov_env["GCOV_PREFIX"] = coverage_dir
             self._coverage_test_dirs.add(coverage_dir)
 
+        if precompiled:
+            precompiled_binary_path = Path(precompiled)
```

Comment:
Maybe let precompiled always be a Path (add a type hint), and handle it wherever this function is called.

Reply:
Good point